gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/NotificationService.groovy

```diff
@@ -93,7 +94,7 @@ class NotificationService {
 
     Request echoRequest = builder.build();
     try {
-      Response response = okHttpClient.newCall(echoRequest).execute()
+      Response response = okHttpClientProvider.getClient(new DefaultServiceEndpoint("echo", echoEndpoint.url)).newCall(echoRequest).execute()
```

Comment:
This is going to instantiate a new client for every request. This client can be initialized once in the class constructor instead.